### ISSUE TITLE
[Bug]: Incorrect behavior of ignored properties in an embbeded object

### ISSUE THREAD

**Issue by Taco55** [realm/realm-dart#1082]
### What happened?

Ignored properties in Embbeded Objects seem not to reflect changes after they are updated.

### Repro steps

For example:

```
@RealmModel()
class $Example {
  @MapTo('_id')
  @PrimaryKey()
  late ObjectId id;

  @Ignored()
  String test1 = "";

  late $ExampleEmbeddedProperty? exampleEmbeddedProperty;
 }


@RealmModel(ObjectType.embeddedObject)
class $ExampleEmbeddedProperty {

  @Ignored()
  String test2 = "";
} 


example.test1 = "test1";
print(example.test1); // prints "test1"

example.exampleEmbeddedProperty!.test2 = "test2";
print(example.exampleEmbeddedProperty!.test2 ); // only prints "" => INCORRECT

```

### Version

^0.8.0+rc

### What Realm SDK flavor are you using?

MongoDB Atlas (i.e. Sync, auth, functions)

### What type of application is this?

Flutter Application

### Client OS and version

MacOS 12.6

**Comment by nielsenko** [comment]
@Taco55 I'm afraid this is expected. Whenever you access a _managed_ realm object (embedded or not) via a link from another realm object, you get a new instance. You should not store state in ignored properties and expect to be able to read them back out like this.

Ignored properties are mostly for caching values that can be inferred from the persisted ones.

**Comment by Taco55** [comment]
@nielsenko Thanks for your answer. I used ignored properties to store temporary data for filtering purposes. Since I use a lot of embedded properties it seemed a logical place to store the related data here as well. I have solved it now by using separate state management.
However, I don't see any use case for non-data-intensive computing since computed properties should work well. May be the documentation can be optimized to clarify proper use of ignored properties?  

The issue can be closed.
